src/definitions.jl

```diff
@@ -1,8 +1,11 @@
 # This file was formerly a part of Julia. License is MIT: https://julialang.org/license
 
-using Base.LinAlg: BlasReal
+using Compat
+using Compat.LinearAlgebra
+using Compat.LinearAlgebra: BlasReal
+import Compat.LinearAlgebra: A_mul_B!, A_ldiv_B!
```

Comment:
These are deprecated in 0.7. We'll have to redesign the API to use `mul!`.

Reply:
Ah right, that change maybe was a bit too mechanical. But for compatibility with both Julia 0.6 and Julia 0.7, we need both `A_mul_B!`/`A_ldiv_B!` and `mul!`/`ldiv!`, right? I'll give it a sot.

Reply:
Julia 0.6 use `Base.fft`

Reply:
Since FFTW doesn't support 0.6 it might be okay to just move to 0.7 only here as well. You could put the Julia lower bound at 0.7.0-DEV.3204, the commit that removed `A_mul_B` et al., then just do everything in terms of `mul!`.

Reply:
Ah yes, good point. Would also allow getting rid of all the Compat stuff. Yes, sounds good.

Reply:
We'll still need Compat for a lot of the stdlib stuff though, since a lot of that came after the `A_mul_B` deprecation.

Reply:
I took the easy way out and set the minimum version to 0.7.0-DEV.3449, that should indeed make Compat unnecessary.